Change event currency to use a choice attribute

emersonmde opened the pull request:
Fixes #446 

Uses a choice attribute so the event currency will be a 3 character ISO currency. Also, adds pycountry to the requirements which provide the list of countries and currencies.

raphaelm commented on src/pretix/base/models/event.py:
```diff
@@ -81,8 +81,11 @@ class Event(LoggedModel):
     live = models.BooleanField(default=False, verbose_name=_("Shop is live"))
     permitted = models.ManyToManyField(User, through='EventPermission',
                                        related_name="events", )
+    CURRENCY_CHOICES = [(settings.CURRENCIES[i].alpha_3, settings.CURRENCIES[i].alpha_3 + " - " + settings.CURRENCIES[i].name)
```
Two comments here. First, can you move that further above whithin this class? I don't think it's usual to mix constants and fields. Best put it before all fields and seperated with an empty line from the fields.

Second, iterating over ``range(0, len(…))`` is not really idiomatic python when we can have
```python
CURRENCY_CHOICES = [(c.alpha_3, c.alpha_3 + " - " + c.name) for c in settings.CURRENCIES]
```
instead.

raphaelm commented on src/requirements/production.txt:
```diff
@@ -37,3 +37,4 @@ pycparser==2.13  # https://github.com/eliben/pycparser/issues/147
 chardet>=2.3,<3
 mt-940==3.2
 vobject==0.9.*
+pycountry
```
Please add this also to the required packages in ``setup.py`` :)

emersonmde commented:
I added 2 tests to ensure that currency symbols and non-ISO currencies aren't able to be used. I haven't used pytest before so hopefully, I did this correctly.

Also updated setup.py and fixed CURRENCY_CHOICES as suggested.

Thank you so much for the feedback! I can usually find a solution in python but it's great to learn more about these best practices. Please let me know if there's anything else you think I should change.

emersonmde commented:
I'm not quite sure why it's saying "All checks have failed". When I do:
> flake8 .
> isort -c -rc .
> python manage.py check
> py.test


From the [Developers Guide](https://docs.pretix.eu/en/latest/development/setup.html) it says all of the tests pass.

rixx commented:
It shows "All checks have passed" now, maybe it just took some time to rebuild?

raphaelm commented:
This seems totally fine to me. Thank you very much for your contribution, it's highly appreciated!
